spec/event_store/sink_spec.rb

```diff
@@ -174,85 +174,88 @@ def persisted_tuples_for_aggregate(aggregate_id)
-
-      it "ensures events are sunk sequentially by locking the database" do
-        t1 = Thread.new {
-          sinker = described_class.new(database: d1, event_type_resolver: event_type_resolver, logger: logger_1)
+      let(:d1_transaction_sleep) { 0.4 }
+      let(:t1) {
+        Thread.new {
           Thread.current.report_on_exception = false # Don't double report RSpec failures
-          sinker.sink([build_staged_event(aggregate_id: aggregate_id_1)])
+
+          sink = described_class.new(database: d1, event_type_resolver: event_type_resolver, logger: logger_1)
+
+          d1.transaction do
+            sink.sink([build_staged_event(aggregate_id: aggregate_id_1)])
+
+            # Sleep inside the transaction so we hold onto the lock longer.
+            sleep d1_transaction_sleep
+          end
 
           # Aggregate 2 should not be saved yet because it doesn't have a lock
           expect(database[:events].select_map(:aggregate_id)).not_to include(aggregate_id_2)
         }
+      }
+      let(:t2) {
+        Thread.new {
+          Thread.current.report_on_exception = false # Don't double report RSpec failures
+
+          # Sleep to ensure this thread gets the lock last, but less than the
+          # sleep inside the transaction so that we try to get the lock while
+          # the other lock is held.
+          sleep d1_transaction_sleep / 2
 
-        t2 = Thread.new {
-          sinker = described_class.new(database: d2, event_type_resolver: event_type_resolver, logger: logger_2)
-          sleep 0.1 # Ensure this thread gets the lock last
-          sinker.sink([build_staged_event(aggregate_id: aggregate_id_2)])
+          sink = described_class.new(database: d2, event_type_resolver: event_type_resolver, logger: logger_2)
+
+          sink.sink([build_staged_event(aggregate_id: aggregate_id_2)])
         }
+      }
 
+      # Expectation is in the t1 thread.
+      def run_threads_with_expectation
         [t1, t2].each(&:join)
+      end
 
-        expect(d1.__try_lock_count).to be < d2.__try_lock_count
+      it "ensures events are sunk sequentially by locking the database" do
+        run_threads_with_expectation
 
-        expect(logger_1).to_not have_received(:info)
-        expect(logger_2).to have_received(:info).with(
-          msg: "event_framework.event_store.sink.retry",
-          tries: an_instance_of(Integer),
-          correlation_id: metadata.correlation_id
-        ).at_least(:once)
+        expect(database[:events].select_map(:aggregate_id)).to match [aggregate_id_1, aggregate_id_2]
       ensure
+        disconnect_other_database_connection
+      end
+
+      context "hitting the lock timeout" do
+        before do
+          lock_timeout = d1_transaction_sleep + 0.1
+          lock_timeout_milliseconds = (lock_timeout * 10).round
+          stub_const("EventFramework::EventStore::Sink::LOCK_TIMEOUT_MILLISECONDS", lock_timeout_milliseconds)
```

Comment:
Could we avoid the const stubbing if we made the lock timeout injectable like the other 2 settings it's replacing?

Reply:
I was going to say “I don't see it needing to be configured because it's feels like a way more solid solution” to your other comment. But this is a good reason to make it configurable from the get go, I'll do it.